pkg/segment/query/processor/statscommand_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright (c) 2021-2024 SigScalr, Inc.
+//
+// This file is part of SigLens Observability Solution
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package processor
+
+import (
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/siglens/siglens/pkg/segment/query/iqr"
+	"github.com/siglens/siglens/pkg/segment/structs"
+	"github.com/siglens/siglens/pkg/segment/utils"
+	"github.com/stretchr/testify/assert"
+)
+
+func getTestData() map[string][]utils.CValueEnclosure {
+	knownValues := map[string][]utils.CValueEnclosure{
+		"col1": {
+			{Dtype: utils.SS_DT_STRING, CVal: "c"},
+			{Dtype: utils.SS_DT_STRING, CVal: "a"},
+			{Dtype: utils.SS_DT_STRING, CVal: "b"},
+			{Dtype: utils.SS_DT_STRING, CVal: "e"},
+			{Dtype: utils.SS_DT_STRING, CVal: "a"},
+			{Dtype: utils.SS_DT_STRING, CVal: "b"},
+		},
+		"col2": {
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(1)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(2)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(3)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(4)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(5)},
+			{Dtype: utils.SS_DT_SIGNED_NUM, CVal: int64(6)},
+		},
+		"col3": {
+			{Dtype: utils.SS_DT_STRING, CVal: "z"},
+			{Dtype: utils.SS_DT_STRING, CVal: "y"},
+			{Dtype: utils.SS_DT_STRING, CVal: "x"},
+			{Dtype: utils.SS_DT_STRING, CVal: "w"},
+			{Dtype: utils.SS_DT_STRING, CVal: "v"},
+			{Dtype: utils.SS_DT_STRING, CVal: "u"},
+		},
+	}
+
+	return knownValues
+}
+
+func getGroupByProcessor() *statsProcessor {
+	measureOperations := []*structs.MeasureAggregator{
+		&structs.MeasureAggregator{
+			MeasureCol:  "col2",
+			MeasureFunc: utils.Count,
+		},
+		&structs.MeasureAggregator{
+			MeasureCol:  "col2",
+			MeasureFunc: utils.Sum,
+		},
+		&structs.MeasureAggregator{
+			MeasureCol:  "col2",
+			MeasureFunc: utils.Avg,
+		},
+	}
+
+	groupByCols := []string{"col1", "col3"}
```

Comment:
Each value in `col3` is unique, so every bucket only has one element. That's a good test, but you should also have a test where multiple rows get aggregated into the same bucket

Reply:
I also have that version of test. Where I have verified that [here](https://github.com/siglens/siglens/pull/1771/files/c52fed5a6e74f531a150d72dd24796f24211f728..8bc6199e79c86e0380e3ce60f07b6d878378350a#diff-a26b93772dc6a4edcf317c1443b994b9eb1d2c38fee43b9717bc15c15a3cbadaR222-R223)

For the second iqr, I removed `col3` and executed the Process, which would make the RRCs fall into the same bucket because `col1` is there and `col3` is filled with default backfill value.